pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -756,7 +757,7 @@ type InMemoryActionSets struct {
 }
 
 // NewActionSetService returns a new instance of InMemoryActionSetService.
-func NewActionSetService(a *acimpl.AccessControl) ActionSetService {
+func NewActionSetService(a *acimpl.AccessControl) *InMemoryActionSets {
```

Comment:
```suggestion
func NewActionSetService(a *acimpl.AccessControl) *InMemoryActionSets {
```

what is the difference here when we return the pointer to a struct?

I think we changed this previously to a interface, if i am not mistaken

Reply:
Yes, there's a slight difference with what wire supports, and I couldn't find an easy way around it. I'm planning to change this back once this PR is merged and I can do a follow up PR to clean up the [action resolvers](https://github.com/grafana/grafana/pull/86801), as we won't need those anymore. So this is just a temporary change. Sorry for making it briefly messy 🙈

Reply:
nada! just wanted to understand what this change was about and if i was missing something.